### PR TITLE
fixing config gen verify flow

### PR DESCRIPTION
**Pull request by SahithiMy**
### :nut_and_bolt: Description: What code changed, and why?
Getting error for config gen which is using verify function in the flow.

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-6308
### :+1: Definition of Done
Fixed the error for config gen verify flow
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Comment by SahithiMy** [comment]
Pipelines:
https://buildkite.com/chef/chef-automate-main-verify-private/builds/32466
https://buildkite.com/chef-oss/chef-automate-main-verify/builds/31188

File: lib/config/hadeployconfigverify_test.go [hunk 1]
```diff
@@ -83,6 +83,41 @@ func TestParseAndVerify(t *testing.T) {
...
+		},
```
Comment (ArvinthC3000) [inline lib/config/hadeployconfigverify_test.go:120]: Please add test cases for the following success scenarios:
- When the location is GCS
- When the location is S3
Error cases:
- When location is not mentioned but backup_config type is "object_storage"

File: lib/config/hadeployconfigverify.go [hunk 2]
```diff
@@ -14,6 +19,12 @@ func (c *HaDeployConfig) Verify() error {
 
 	// Validate ExistingInfra (on prem)
 	if c.IsExistingInfra() {
+		if c.ObjectStorage != nil && c.ObjectStorage.Config.Location == GCS_STORAGE {
+			err := c.readGCSConfigFile(c.ObjectStorage.Config)
+			if err != nil {
+				return err
```
Comment (ArvinthC3000) [inline lib/config/hadeployconfigverify.go:25]: Can we move this into the following function in line no - 191: 
https://github.com/chef/automate/blob/e2fbc14136df0e5472c24e5702e6a48f82079bc6/lib/config/hadeployconfigverify.go#L191C11-L191C11

Function name and signature: verifyObjectStorage(objectStorage *ConfigObjectStorage) error

**Review by ArvinthC3000 — CHANGES_REQUESTED** [review]

File: lib/config/hadeployconfigverify.go [hunk 3]
```diff
@@ -14,6 +19,12 @@ func (c *HaDeployConfig) Verify() error {
 
 	// Validate ExistingInfra (on prem)
 	if c.IsExistingInfra() {
+		if c.ObjectStorage != nil && c.ObjectStorage.Config.Location == GCS_STORAGE {
+			err := c.readGCSConfigFile(c.ObjectStorage.Config)
```
Comment (SanjuPal01) [inline lib/config/hadeployconfigverify.go:23]: We are reading the GcsServiceAccount.json here then do we need the changes which made in Parse function (lib/config/hadeployconfig.go)? Here also we are reading the file. and then invoking the same Verify function and now in verify function we are handling it.
Same for Post Deployment: https://github.com/chef/automate/pull/8176 we have to remove all these changes right?

**Review by Dmaddu — APPROVED** [review]

**Pull request merged**